core/health/impl/src/main/java/org/opennms/core/health/impl/DefaultHealthCheckService.java

```diff
@@ -95,8 +93,8 @@ public DefaultHealthCheckService(BundleContext bundleContext) {
         try {
             // Fail if no checks are available
             List<HealthCheck> checks = getHealthChecks();
-            if (tag != null){
-                checks = checks.stream().filter(check -> check.getTags().contains(tag)).collect(Collectors.toList());
+            if (tags != null && tags.stream().anyMatch(tag -> !Strings.isNullOrEmpty(tag))){
+                checks = checks.stream().filter(check -> check.getTags().stream().anyMatch(tags::contains)).collect(Collectors.toList());
             }
             if (checks == null || checks.isEmpty()) {
```

Comment:
checks getting accessed before null check.

Reply:
The checks.stream actually can handle the null value. But I'll add test cases after all design works were done and when we are sure no more further code changes needed.

Reply:
checks == null seems unnecessary.  I'm good with implementation so far.  Also update `health-check` on Karaf.
Add `-t` option on `HealthCheckCommand`

Reply:
Sure. I've wrote up https://issues.opennms.org/browse/NMS-13327 for this.